linux/mcompile/keymap/filesystem.cpp

```diff
@@ -0,0 +1,188 @@
+
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#include <emscripten/bind.h>
+#endif
+
+#include <cassert>
+#include <algorithm>
+#include <iostream>
+#include "filesystem.h"
+
+#ifdef _MSC_VER
+#include <io.h>
+#else
+#include <unistd.h>
+#endif
+
+//#define _DEBUG_FOPEN  1
+
+#ifdef __EMSCRIPTEN__
+
+const std::string get_wasm_file_path(const std::string& filename) {
+  // Verify that we are passing a fully-qualified path
+  // TODO: we need to support relative paths based on CWD
+#if _DEBUG_FOPEN
+  std::cout << "get_wasm_file_path ENTER (" << filename << ")" << std::endl;
+#endif
+
+  assert(
+    (filename.length() > 0 && filename.at(0) == '/') ||
+    (filename.length() > 1 && filename.at(1) == ':')
+  );
+
+#if _DEBUG_FOPEN
+  std::cout << "get_wasm_file_path assert passed " << std::endl;
+#endif
+
+  EM_ASM_({
+    //console.log('EM_ASM enter');
+    let path = Module.UTF8ToString($0);
+    const isWin32Path = path.match(/^[a-zA-Z]:/);
+
+    //console.log('EM_ASM path = '+path);
+    let root = '/nodefs-mount';
+    if(!FS.analyzePath(root).exists) {
+      //console.log('EM_ASM mkdir '+root);
+      FS.mkdir(root);
+      if(!isWin32Path) {
+        //console.log('EM_ASM mount '+root);
+        FS.mount(NODEFS, {root : '/'}, root);
+      }
+    }
+
+    if(isWin32Path) {
+      // Win32 path, one mount per drive
+      root += "/" + path.charAt(0);
+      if(!FS.analyzePath(root).exists) {
+        //console.log('EM_ASM mkdir '+root);
+        FS.mkdir(root);
+        //console.log('EM_ASM mount '+root);
+        FS.mount(NODEFS, {root : path.substr(0,3) }, root);
+      }
+    }
```

Comment:
```suggestion
    if(isWin32Path) {
      // Win32 path, one mount per drive
      root += "/" + path.charAt(0);
      if(!FS.analyzePath(root).exists) {
        //console.log('EM_ASM mkdir '+root);
        FS.mkdir(root);
        //console.log('EM_ASM mount '+root);
        FS.mount(NODEFS, {root : path.substr(0,3) }, root);
      }
    } else {
		//console.log('EM_ASM mount '+root);
        FS.mount(NODEFS, {root : '/'}, root);		
	}
```

Reply:
OK-addressed  -> #11887
filesystem is removed from mcompile-linux 
function Open_File is moved to the module mc_kmxfile